### DISCUSSION TITLE
Is there a production example of oak?

### DISCUSSION THREAD

**Discussion by viztastic** [oakserver/oak#175] [General]
It'd be great to include a lean example of how Oak could/should be configured for a production use case. Helps eliminate best-practice guesswork especially with respect to security and approaches to testing.

**Comment by asad-devx** [comment]
Here is my starter repo for production API with oak. Still, many things need to add or update. 
https://github.com/asad-mlbd/deno-api-starter-oak

Expecting Comments, PR, issues, thanks!

**Comment by viztastic** [comment]
Hi @asad-mlbd thanks for sharing, I took a look and posted a few questions/thoughts in your repo. I think it's a great start but there's a few (critical) bits and pieces before it can be considered a production/security-ready codebase.

But the idea of having something like this is amazing.

**Comment by asad-devx** [comment]
> * Any chance you could include an example of how to connect to MongoDB in a production friendly manner.
Currently, I have some other plans like integrating testing, work with cloud deployment, etc.

To make it work with MongoDB, just have to change `db.ts`, and the repository layer. I can do that in future, or if someone interested can give a PR, I can keep a separate branch for that. Are you interested?

> * Noticed you're storing password in code, pretty sure that's bad security practice across the board? Shouldn't we be using something like docker secrets or chamber to handle this?

Actually that is `env.example`. And in README it is mentioned to change the pass and JWT secrets. Kept the sample pass and Jwt values in example file so that anyone can just rename it and can run the project easily.

> * Also nervice about storing any type of secret in the codebase (jwt secrets included)

I prefer to keep it in env variable

> * Shouldn't there be some sort of listener for the SIGKILL, SIGINT and similar events to handle db cleanup before shut-down?

Will check, can you give some related examples, please?

> * Would be great if there's some sort of guidance/pattern for how to store data in a produciton-friendly manner, it seems to just mount to a directly on the host computer?

In future, I might write some posts regarding prod deploy. 

Thanks @viztastic ! Please keep posting your comments,

**Comment by kitsonk** [comment]
Let's keep the conversation here on topic.  Mentioning here that you commented on other issues in other repos really isn't on topic.

**Comment by GuilhermeVendramini** [comment]
Hi guys, I'm creating a small REST API and CMS using AOK:

Project: **http://denocms.net/**

Demo online: **http://demo.denocms.net/**
The demo will be slow at the first access, because I'm hosting on Heroku free account.

In the demo you can create an user, login and create contents to test the application (all content and users will be removed after a period).

This is my first experience with development using a javascript framework, so feedbacks are welcome.
